config/rbac/rbac_role.yaml

```diff
@@ -42,3 +42,11 @@ rules:
   - update
   - patch
   - delete
+- apiGroups:
```

Comment:
Can we scope this to something less than _all_ secrets in _all_ namespaces?

Reply:
Also it is not safe to edit this file, because controller-tools (https://github.com/kubernetes-sigs/cluster-api/blob/master/Makefile#L59) would override this.

Reply:
What would be the right way to scope/implement this?

Reply:
Having access to all secrets sounds odd, but I also can't imagine how to solve it differently.

I would like to follow kubebuilders annotations ( `kubebuilder:rbac: ...`, see https://github.com/kubernetes-sigs/cluster-api/blob/299c318690e526b7d8c704ab40a813c00ab7b0b0/pkg/controller/node/node_controller.go#L80) and enable `controller-gen` in the Makefile. For sure, you must migrate the current rbac fules to the annotations as well.

Maybe this is a bit out of scope of this PR, and currently running `controller-gen` doesn't modify the `rbac_role.yaml` (as far as I see). Should we maybe just create an issue to migrate rbacs to kubebuilder?

Reply:
I agree this is out of scope for this PR. I'd rather get this in and improve it in a follow-up patch. I'd like this patch to focus on improving the UX/functionality and we can clean it up asap.